ledger-api/grpc-definitions/com/daml/ledger/api/v1/completion.proto

```diff
@@ -29,4 +31,50 @@ message Completion {
   // Must be a valid LedgerString (as described in ``value.proto``).
   // Optional
   string transaction_id = 3;
+
+  // The application ID that was used for the submission, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  string application_id = 4;
+
+  // The set of parties on whose behalf the commands were executed.
+  // Contains the union of ``party`` and ``act_as`` from ``commands.proto``.
+  // The order of the parties need not be the same as in the submission.
+  // Each element must be a valid PartyIdString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  repeated string act_as = 5;
+
+  // The submission ID this completion refers to, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
```

Comment:
@SamirTalwar-DA Isn't it optional for an application to set it in the submission?

Reply:
I think it is set by the participant in that case, though. Is that right?

Reply:
Checked, the participant or the committer, so the participant will have it (starting with a certain SDK). The comment is correct.